code/game/objects/compass/compass_overmap.dm

```diff
@@ -0,0 +1,85 @@
+#define HEADER_SPEED_THRESHOLD 0.35
+
+/obj/compass_holder/overmap
+	compass_interval = 9
+	var/image/compass_heading_marker
+	var/obj/machinery/computer/ship/helm/owner
+
+/obj/compass_holder/overmap/Initialize(mapload)
+
+	owner = loc
+
+	if(!istype(owner))
+		PRINT_STACK_TRACE("Overmap compass initialized in non-helm atom: [type]")
+		return INITIALIZE_HINT_QDEL
+
+	compass_heading_marker = new /image/compass_marker
+	compass_heading_marker.maptext = STYLE_SMALLFONTS("<center>△</center>", 7, COLOR_WHITE)
+	compass_heading_marker.filters = filter(type="drop_shadow", color = "[owner.linked.color]aa", size = 2, offset = 1,x = 0, y = 0)
+	compass_heading_marker.layer = HUD_BASE_LAYER
+	compass_heading_marker.plane = HUD_PLANE
+	compass_heading_marker.color = owner.linked.color
```

Comment:
What if linked is null? It's possible it's fine, but I can't tell. Same for other procs below.

Reply:
Null color is valid, not sure what you mean?

Reply:
if the `linked` var in this line is null, this will runtime. It's unclear to me from code inspection if this is possible, and want to make sure you are certain it's not.

Reply:
Gotcha, added a check.